test/func_ds3_share_open_access.py

```diff
@@ -282,6 +288,52 @@ def _run_all(self, fstype, tests, testtype):
         ret = carry ? -1 : 0;
         break;
       }
+
+      case SETATT: {
+        /*
+          DOS 2+ - CHMOD - SET FILE ATTRIBUTES
+        */
+
+        const char *fname = FN1 "." FE1;
+        unsigned int oattr, nattr;
+        int rc;
+
+        rc = _dos_getfileattr(fname, &oattr);
+        if (rc != 0) {
+          printf("FAIL:('%s', '%s', '%s')[secondary getattr1 failed(%d)]\n",
+            argv[2], argv[3], argv[4], rc);
+          return -1;
+        }
+        if (oattr & _A_HIDDEN) {
+          printf("FAIL:('%s', '%s', '%s')[secondary getattr1 invalid(0x%02x)]\n",
+            argv[2], argv[3], argv[4], oattr);
+          return -1;
+        }
+
+        rc = _dos_setfileattr(fname, oattr | _A_HIDDEN);
+        if (rc != 0) {
+          printf("INFO:('%s', '%s', '%s')[secondary setattr failed(%d)]\n",
+            argv[2], argv[3], argv[4], rc);
+        } else {
+          int rc2;
+
+          printf("INFO:('%s', '%s', '%s')[secondary setattr success]\n",
+            argv[2], argv[3], argv[4]);
+
+          rc2 = _dos_getfileattr(fname, &nattr);
+          if (rc2 != 0) {
+            printf("INFO:('%s', '%s', '%s')[secondary getattr2 failed(%d)]\n",
+              argv[2], argv[3], argv[4], rc2);
+          } else if (nattr != (oattr | _A_HIDDEN)) {
+            printf("INFO:('%s', '%s', '%s')[secondary getattr2 invalid(0x%02x)]\n",
+              argv[2], argv[3], argv[4], nattr);
+          }
+        }
+
+        ret = (rc != 0) ? -1 : 0;
```

Comment:
Is this where error is processed?
But if rc!=0 (setattr failde), shouldn't we
return 0 - success?

Reply:
Just where the logic inversion happens?

Reply:
As this is common code for delete+fcb, delete+path, rename+fcb and rename+path, that line just makes the return values match those of the other tests. Later on we check the arguments for the expected response of ALLOW / DENY to this specific test before examining the return value and posting the final PASS or FAIL text.

Reply:
Line 339 is where the inversion happens.

Reply:
Ah, cool.